### PR TITLE
001 Create New Delivery Employee API

### PR DESCRIPTION
**Pull request by chrisknos**

### CODE REVIEW

File: src/main/java/org/example/api/DeliveryEmployeeService.java [hunk 1]
```diff
@@ -25,12 +27,28 @@ public DeliveryEmployeeService(DeliveryEmployeeDAO dao){
...
-        String error = employeeValidator.isValidEmployee(deliveryEmployee);
-        if(error != null){
-            throw new ValidationFailedException(error);
+//    public int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployee) throws FailedToCreateException, ValidationFailedException {
+//        String error = employeeValidator.isValidEmployee(deliveryEmployee);
+//        if(error != null){
+//            throw new ValidationFailedException(error);
+//        }
+//        return dao.createDeliveryEmployee(deliveryEmployee);
+//    }
```
Comment (shaunganley) [inline src/main/java/org/example/api/DeliveryEmployeeService.java]: remove if not needed

File: src/main/java/org/example/db/DeliveryEmployeeDAO.java [hunk 2]
```diff
@@ -13,15 +14,15 @@
...
+    static DatabaseConnector databaseConnector = new DatabaseConnector();
 
     /**
      * inserts a new employee into employee table, then adds employee to delivery employee table via employee id
      * @param deliveryEmployee
      * @return id of created employee in database
      * @throws SQLException
      */
-    public int createDeliveryEmployee(DeliveryEmployee deliveryEmployee) throws FailedToCreateException {
+    public static int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployee) throws FailedToCreateException {
```
Comment (shaunganley) [inline src/main/java/org/example/db/DeliveryEmployeeDAO.java]: this shouldn't be static

File: src/main/java/org/example/db/DeliveryEmployeeDAO.java [hunk 3]
```diff
@@ -13,15 +14,15 @@
 public class DeliveryEmployeeDAO {
 
     // create instance of database connector class
-    DatabaseConnector databaseConnector = new DatabaseConnector();
+    static DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (shaunganley) [inline src/main/java/org/example/db/DeliveryEmployeeDAO.java]: this shouldn't be static. if you wanted to access it statically you wouldn't need this property

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/api/DeliveryEmployeeService.java [hunk 4]
```diff
@@ -26,12 +28,22 @@ public DeliveryEmployeeService(DeliveryEmployeeDAO dao){
...
      */
-    public int createDeliveryEmployee(DeliveryEmployee deliveryEmployee) throws FailedToCreateException, ValidationFailedException {
-        String error = employeeValidator.isValidEmployee(deliveryEmployee);
-        if(error != null){
-            throw new ValidationFailedException(error);
+
+    public int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployee) throws FailedToCreateDeliveryEmployeeException {
+        DeliveryEmployeeDAO deliveryEmployeeDAO = new DeliveryEmployeeDAO();
+        try {
+            int id = deliveryEmployeeDAO.createDeliveryEmployee(deliveryEmployee);
```
Comment (shaunganley) [inline src/main/java/org/example/api/DeliveryEmployeeService.java:45]: no validation? :(

**Review by shaunganley — CHANGES_REQUESTED** [review]
needs validation

File: src/main/java/org/example/api/DeliveryEmployeeService.java (hunk 1 above)
Comment (chrisknos) [reply src/main/java/org/example/api/DeliveryEmployeeService.java]: sorted 👍

File: src/main/java/org/example/db/DeliveryEmployeeDAO.java (hunk 2 above)
Comment (chrisknos) [reply src/main/java/org/example/db/DeliveryEmployeeDAO.java]: Sorted 👍

File: src/main/java/org/example/db/DeliveryEmployeeDAO.java (hunk 3 above)
Comment (chrisknos) [reply src/main/java/org/example/db/DeliveryEmployeeDAO.java]: Sorted 👍

File: src/main/java/org/example/api/DeliveryEmployeeService.java (hunk 4 above)
Comment (chrisknos) [reply src/main/java/org/example/api/DeliveryEmployeeService.java:45]: Validation added 👍

**Review by shaunganley — APPROVED** [review]

**Pull request merged**